[Yew Wei Quan] iP

weiquany opened the pull request:
Duke
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))
Duke frees your mind from having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ *SUPER* FAST to use

All you need to do is,

1. download it from [here](https://www.youtube.com/watch?v=djV11Xbc914).
2. something
3. something

And it is **FREE**!
 
Features:
- [X] Managing tasks
- [ ] something new (coming soon)
- [ ] something new (coming soon)
___
Here's the `main` method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

weiquany committed:
Implement greet, echo, exit in Duke

weiquany committed:
Add TaskList, and add, list function in Duke

weiquany committed:
Implement tasks done status

weiquany committed:
Add ToDo, Event, Deadline

weiquany committed:
Add UI tests

weiquany committed:
Refactor commands from Duke to individual classes

weiquany committed:
Implement error handling

weiquany committed:
Update tests

weiquany committed:
Add delete task option and add delete test

weiquany committed:
Move scanner to Ui class

weiquany committed:
Fix switch cases identation

weiquany committed:
Implement command parser error handling

weiquany committed:
Add saving and loading tasks to and from storage

weiquany committed:
Rename CommandParser back to Parser

weiquany committed:
Add dates instead of using string

weiquany committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Parser.java
#	src/main/java/Task.java

weiquany committed:
Refactor Command classes

weiquany committed:
Organize codebase into packages

weiquany committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2223S1/ip

weiquany committed:
Add gradle support

weiquany committed:
Add JUnit tests for TaskList and Parser

weiquany committed:
Update error messages

weiquany committed:
Add JavaDoc comments to all classes

weiquany committed:
Tweak code to comply with coding standard

weiquany committed:
Add FindCommand

weiquany committed:
Merge branch 'branch-A-JavaDoc'

weiquany committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/task/Event.java
#	src/main/java/duke/util/Parser.java
#	src/main/java/duke/util/Storage.java
#	src/main/java/duke/util/Ui.java

weiquany committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/task/Task.java
#	src/main/java/duke/task/TaskList.java
#	src/main/java/duke/util/Parser.java

weiquany committed:
Add JavaDoc comments for search related functions

weiquany committed:
Rename variables and add some JavaDoc comments

weiquany committed:
Add JavaFx GUI

weiquany committed:
Add FXML

weiquany committed:
Update Javadoc comments

weiquany committed:
Add welcome message

weiquany committed:
Merge branch 'branch-Level-10'

weiquany committed:
Use var-args

weiquany committed:
Merge branch 'branch-A-Varargs'

# Conflicts:
#	src/main/java/duke/util/Ui.java

weiquany committed:
Fix dialog ellipsis

yeojunjie commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,71 @@
+ * Event task.
+ */
+public class Event extends Task {
+    private LocalDate at;
```
Hello Wei Quan, I think it would be clearer to use a noun as a variable name, as the preposition 'at' could also be used in front of a location.
```suggestion
    private LocalDate startTime;
```

rylzxc replied:
Yup, I agree with the above comment.

rylzxc replied:
I agree with this and the similar comment made on the deadline class

yeojunjie commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,93 @@
+     * Returns a String representation of the {@code Task} in storage format.
+     */
+    public String getStorageFormat() {
+        return this.getStatusBit() + " | " + this.description;
```
Hi Wei Quan, the method name `getStorageFormat` effectively communicates that this method is meant to encode a `Task` as a `String` for storage. Other authors I reviewed have used names such as `stringify`, which is problematic because it does not self-explain the difference between `stringify` and `toString`.

However, I think it would be clearer to use `String.format()` here. It might be unnecessary to use `getStatusBit()`.

```suggestion
        return String.format("%d | %s", this.isDone ? 1 : 0, this.description);
```

yeojunjie commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,163 @@
+     *
+     * @return {@code TaskList} of all tasks containing the search keyword in their description.
+     */
+    public TaskList find(String keyword) {
```
Hi Wei Quan, I thought that the naming conventions were well-followed in this method -- the method name `get` is a verb, the temporary variable `tasksMatched` and the parameter `keyword` is self-explanatory.

yeojunjie commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,72 @@
+ * Task with a deadline.
+ */
+public class Deadline extends Task {
+    private LocalDate by;
```
Hello Wei Quan, the field `by` could be renamed to the noun `dueDate` to better explain what it represents. Otherwise, `by` could also potentially refer to the person who added this task.

```suggestion
    private LocalDate dueDate;
```

yeojunjie commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,71 @@
+     */
+    @Override
+    public String toString() {
+        return "[E]" + super.toString() + " (at: " + this.getFormattedDateString() + ")";
```
Once again, I think using `String.format()` would make it easier to see where whitespace will end up in the display format.

```suggestion
        return String.format("[E]%s (at: %s)", super.toString(), this.getFormattedDateString());
```

yeojunjie reviewed (commented):
Hello Wei Quan, I have left some comments regarding code quality guidelines!

rylzxc commented on src/main/java/duke/util/Parser.java:
```diff
@@ -0,0 +1,148 @@
+            args = commandAndArguments[1].strip();
+        }
+
+        switch (inputCommand) {
+        case "help":
+            return new HelpCommand();
+        case "bye":
+            return new ByeCommand();
+        case "list":
+            return new ListCommand();
+        case "mark":
+            verifyArgumentsIsNotEmpty(args);
+            return new MarkCommand(parseTaskIndex(args));
+        case "unmark":
+            verifyArgumentsIsNotEmpty(args);
+            return new UnmarkCommand(parseTaskIndex(args));
+        case "todo":
+            verifyArgumentsIsNotEmpty(args);
+            return new AddCommand(parseTodoArguments(args));
+        case "event":
+            verifyArgumentsIsNotEmpty(args);
+            return new AddCommand(parseEventArguments(args));
+        case "deadline":
+            verifyArgumentsIsNotEmpty(args);
+            return new AddCommand(parseDeadlineArguments(args));
+        case "delete":
+            verifyArgumentsIsNotEmpty(args);
+            return new DeleteCommand(parseTaskIndex(args));
+        case "find":
+            verifyArgumentsIsNotEmpty(args);
+            return new FindCommand(args);
+        case "":
+            return new EmptyCommand();
+        default:
+            return new UnknownCommand();
+        }
```
Hi Wei Quan, I don't think this is a coding standard violation but I was wondering if it is better to declare constants for your respective cases? I suggest this because if you're managing a super large codebase and you somehow have to check for these `inputCommands` , a constant may save you the time taken to find these hard-coded commands. 

Let me know your thoughts! :)

rylzxc commented on src/test/java/duke/task/TaskListTest.java:
```diff
@@ -0,0 +1,121 @@
+
+import duke.DukeException;
+
+public class TaskListTest {
```
I like your well thought out test cases in your test classes!

rylzxc commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,72 @@
+/**
+ * Task with a deadline.
+ */
+public class Deadline extends Task {
```
Hi, again, not a coding standard violation but I was wondering if adding Task behind the respective classes for tasks will be better? I feel it allows readers to easily navigate to the type of task they want to see. Also, Event could be misleading as it can represent things like button onClick events etc

```suggestion
public class DeadlineTask extends Task {
```

rylzxc reviewed (commented):
Overall, clean code, simple to understand.

weiquany committed:
Add assertions

Assertions are helpful in checking for regressions.

Let's add assertions continuously to ensure that bugs in the program
are caught.

weiquany committed:
Replace magic constants with symbolic constants

Magic constants do not have an obvious meaning, making it hard
to understand the program.

Let's avoid introducing magic constants and use symbolic
constants instead.

weiquany committed:
Merge pull request #1 from weiquany/branch-A-Assertions

Add assertions

weiquany committed:
Merge branch 'master' into branch-A-CodeQuality

weiquany committed:
Merge pull request #2 from weiquany/branch-A-CodeQuality

Replace magic constants with symbolic constants

weiquany committed:
Add continuous integration

weiquany committed:
Add space after comma

weiquany committed:
Add friendlier syntax

weiquany committed:
Update build.gradle

weiquany committed:
Add text bubble for UI

weiquany committed:
Resize UI height

weiquany committed:
Update README

weiquany committed:
Resize UI height
